spaces.go

```diff
@@ -0,0 +1,413 @@
+type rawSpace struct {
+	D []Event `json:"d"`
+}
+
+// Space stores a set of Events, all belonging to the same location ("Space"), such as TR1 or CTPH.
+type Space []Event
+
+// Spaces contains all info about Spaces. Use spaces.Restrict(...) to select a subset to be displayed.
+type Spaces []Space
+
+// ByStartDate implements sort.Interface for Space based on the space[i].Start field
+type ByStartDate Space
+
+func (space ByStartDate) Len() int      { return len(space) }
+func (space ByStartDate) Swap(i, j int) { space[i], space[j] = space[j], space[i] }
+func (space ByStartDate) Less(i, j int) bool {
+	return ParseJSONDate(space[i].Start).Before(ParseJSONDate(space[j].Start))
+}
+
+// SortByStartDate returns a new Space whose Events are sorted in chronological order.
+func (space Space) SortByStartDate() Space {
+	sort.Sort(ByStartDate(space))
+	return space
+}
+
+// ScanRawSpace requests data for a single Space from nususc.com
+func ScanRawSpace(rawSpacePointer *rawSpace, facilityID int) {
+
+	spacesURL := "http://www.nususc.com/USCWebsiteInformationAPI.asmx/GetSpacesBookingRecord"
+	//data to send in body
+	requestBody := []byte(fmt.Sprintf("{ \"facilityID\": \"%d\" }", facilityID))
+
+	//Send http request and receive response
+	client := http.Client{}
+	request, requestErr := http.NewRequest("POST",
+		spacesURL,
+		bytes.NewBuffer(requestBody))
+	if requestErr != nil {
+		panic("Error in spaces.go while creating HTTP request")
+	}
+	request.Header.Set("Content-Type", "application/json")
+
+	response, responseError := client.Do(request)
+	if responseError != nil {
+		panic("Error in spaces.go while receiving HTTP response")
+	}
+
+	defer response.Body.Close() //Close buffer
+
+	//Get raw json data
+	rawData, rawDataError := ioutil.ReadAll(response.Body)
+	if rawDataError != nil {
+		panic("Error in spaces.go while reading HTTP response body")
+	}
+
+	//Unmarshal json data
+	json.Unmarshal(rawData, rawSpacePointer)
+}
+
+// GetSpace obtains Space information for a given space, as identified by its facilityID. (cf. SpaceName(facilityID))
+func GetSpace(facilityID int) Space {
+	rawSpace := rawSpace{}
+	ScanRawSpace(&rawSpace, facilityID)
+	var space Space = rawSpace.D
+	return space
+}
+
+// GetSpaces obtains all spaces booking info from nususc.com/Spaces.aspx
+func GetSpaces() Spaces {
+	spaces := Spaces{}
+	for i := 0; i < 6; i++ {
+		spaces = append(spaces, (GetSpace(i + 1)).SortByStartDate())
+	}
+	return spaces
+}
+
+// ParseJSONDate parses date from format used in nususc.com's API into a time.Time object
+func ParseJSONDate(date string) time.Time {
+	format := "2006-01-02T15:04:05"
+	t, err := time.Parse(format, date)
+	if err != nil {
+		panic(fmt.Sprintf("Error in spaces.go while parsing %s as a JSON date", date))
+	}
+	return t
+}
+
+// ParseDDMMYYDate parses user-inputted dd/mm/yy date into time.Time
+func ParseDDMMYYDate(date string) (time.Time, error) {
+	//Attempt to parse as dd/mm/yy
```

Comment:
This function is really big - break it up.

But more importantly, this looks really hackish. Why do we have so many cases? Can you describe the problem you're solving here?